### DISCUSSION TITLE
Storing a custom dimensional reduction Seurat v5

### DISCUSSION THREAD

**Discussion by mrmxhmd** [satijalab/seurat#8726] [Q&A]
Hello,

I was following this [vignette](https://satijalab.org/seurat/archive/v3.0/dim_reduction_vignette.html) to add a custom dimensional reduction with Seurat v5, however I am encountering an error.  

The table I use is: [Combined_embedding_v0.csv](https://github.com/satijalab/seurat/files/14877709/Combined_embedding_v0.csv)

And here is my code
```
pia <- read.csv("Combined_embedding_v0.csv")
names(pia) <- NULL
colnames(pia)[2:21] <- paste0("pia_", 1:20)
pia <- subset (pia, select = 2:21,)
pia <- data.matrix(pia, rownames.force = NA)
fsMerge[["pia"]] <- CreateDimReducObject(embeddings = pia, key = "pia_", assay = DefaultAssay(fsMerge))
```
This is the error I receive: Error: Cannot add new cells with [[<-

I also tried going through the test data in the vignette and I get this error at the step `d <- dist(t(GetAssayData(pbmc, slot = "scale.data")))`: 
Error in h(simpleError(msg, call)) : 
  error in evaluating the argument 'x' in selecting a method for function 't': no applicable method for 'GetAssayData' applied to an object of class "seurat"

Is this a problem with the version or am I doing something wrong?

Thank you in advance!

**Comment by jsicherman** [comment]
Unsure if this is the now-intended way or not, but you could effectively "override" the `SeuratObject`'s implementation of `[[` by directly indexing into the object's `reductions` slot, like so (note the new `@reductions`):

```
fsMerge@reductions[["pia"]] <- CreateDimReducObject(embeddings = pia, key = "pia_", assay = DefaultAssay(fsMerge))
```
